Configuration/TCA/Overrides/sys_file_reference.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+use TYPO3\CMS\Core\Utility\ExtensionManagementUtility;
+
+$additionalColumns = [
+    'language' => [
+        'label' => 'LLL:EXT:core/Resources/Private/Language/locallang_general.xlf:LGL.language',
+        'description' => 'LLL:EXT:html5mediakit/Resources/Private/Language/locallang_csh_media.xlf:language.description',
+        'config' => [
+            'type' => 'input',
+            'eval' => 'trim,required,alpha,nospace,lower,2',
+            'default' => 'en'
+        ],
+    ],
+];
+
+ExtensionManagementUtility::addTCAcolumns(
+    'sys_file_reference',
+    $additionalColumns
+);
+
+ExtensionManagementUtility::addToAllTCAtypes(
+    'sys_file_reference',
+    'language',
+    '',
+    'after:title'
+);
+
+$GLOBALS['TCA']['sys_file_reference']['columns']['language']['displayCond'] = 'FIELD:fieldname:=:subtitles';
```

Comment:
This should be a [combined](https://docs.typo3.org/m/typo3/reference-tca/main/en-us/Columns/Properties/DisplayCond.html#combining-conditions) `displayCond` that also checks the tablename:

```php
'displayCond' => [
   'AND' => [
       'FIELD:tablenames:=:tx_html5mediakit_domain_model_media',
       'FIELD:fieldname:=:tracks'
   ],
],
```

Also: as far as I can see this can be merged in the `$additionalColumns` definition at the beginning of the file.